onnxruntime/core/session/plugin_ep/ep_api.cc

```diff
@@ -205,6 +210,275 @@ ORT_API(uint64_t, GetSyncIdForLastWaitOnSyncStream, _In_ const OrtSyncStream* pr
   return id;
 }
 
+ORT_API_STATUS_IMPL(CreateKernelRegistry, _Outptr_ OrtKernelRegistry** kernel_registry) {
+  API_IMPL_BEGIN
+  auto unique_kernel_registry = std::make_unique<OrtKernelRegistry>();
+  unique_kernel_registry->registry = std::make_shared<onnxruntime::KernelRegistry>();
+
+  *kernel_registry = unique_kernel_registry.release();
+  return nullptr;
+  API_IMPL_END
+}
+
+ORT_API(void, ReleaseKernelRegistry, _Frees_ptr_opt_ OrtKernelRegistry* kernel_registry) {
+  delete kernel_registry;
+}
+
+ORT_API_STATUS_IMPL(KernelRegistry_AddKernel, _In_ OrtKernelRegistry* kernel_registry,
+                    _In_ const OrtKernelDef* kernel_def, _In_ OrtKernelCreateFunc kernel_create_func,
+                    _In_ void* kernel_create_func_state) {
```

Comment:
should this implementation also be surrounded by API_IMPL_BEGIN and API_IMPL_END?

Reply:
Yes thanks.